src/Python-Wave-Serial-Encoder.py

```diff
@@ -0,0 +1,383 @@
+    else:
+        port.set("Connection Failed")
+        connection.config(bg="red")
+
+
+# Reading
+def read_from_port(q, e, e1, e3, arduino, entry, result):
+    while e.is_set():
+        if not e1.is_set():
+            if path.isfile(entry.get()):
+                os.remove(entry.get())
+
+            try:
+                f = wave.open(entry.get(), "wb")
+                f.setnchannels(1)
+                f.setframerate(16000)
+                f.setsampwidth(2)
+
+                arduino.write(b"\x01")
+                arduino.flushInput()
+                result.set("Writing on " + entry.get())
+
+                while arduino.isOpen():
+                    ser_bytes = arduino.read(256)
+
+                    if ser_bytes != b"":
+                        f.writeframes(ser_bytes)
+                        # print(ser_bytes)
+
+                    if not q.empty():
+                        f.close()
+                        arduino.close()
+                        e1.set()
+            except IOError:
+                result.set("Error: file path not valid")
+        time.sleep(0.1)
+    e3.set()
+
+
+# Start recording
+def start(e1, q, arduino, port, connection, entry, result):
+    # print("Start")
+
+    if entry.get() != "":
+        q.queue.clear()
+        try:
+            if not arduino.isOpen():
+                arduino.open()
+            connection.config(bg="green")
+            port.set("Port open")
+            e1.clear()
+        except serial.SerialException:
+            port.set("Port not select or connection button not pushed")
+            connection.config(bg="red")
+    else:
+        result.set("File name entry empty")
+
+
+# Stop recording
+def stop(q, arduino, port, connection, e1):
+    # print("Stop")
+    if not e1.is_set():
+        arduino.write(b"\x00")
+        arduino.flushInput()
+        a = 0
+        q.put(a)
+        port.set("Port closed")
+        connection.config(bg="black")
+
+
+# Setting save path
+def save(entry):
+    dir_name = filedialog.asksaveasfilename(
+        defaultextension=".wav", filetypes=[("Wave", "*.wav")]
+    )
+    entry.delete(0, "end")
+    entry.insert(0, dir_name)
+    entry.xview_moveto(1)
+
+
+def man_callback():
+    readme = tk.Tk()
+    readme.title("Python Wave Serial Encoder")
+    readme.geometry("560x260")
+    text = tk.Label(readme, text="Manual", font="Helvetica 16 bold")
+    man = tk.Label(
+        readme,
+        text='1. Use the dropdown menu to select the port and \
+        then click on "Connect" \n\n\
+        2. Choose the path where the file will be saved the file \
+        by using the "Save" button\n\n\
+        3. Use start and stop button to respectively start and stop \
+        the recording.\n\n\
+        Please note:  to record more then one file, change the path \
+        else the old file will be overridden.\n\n\
+        In case the dropdown menu does not detect the port, use the button \
+        "Refresh" to update the Serial Port',
```

Comment:
During execution of the script I saw that the text formatting is not correct in the Manual box.
You can change it like this to have a proper display:
```suggestion
        justify=tk.LEFT,
        text="""1. Use the dropdown menu to select the port and then click on "Connect"

2. Choose the path where the file will be saved the file by using the "Save" button

3. Use start and stop button to respectively start and stop the recording.

Please note: to record more than one file, change the path else the old file will be
overridden.

In case the dropdown menu does not detect the port, use the "Refresh" button
to update the Serial Port""",
```